docs/source/tutorials/interactive_simulation.rst

```diff
@@ -0,0 +1,217 @@
+
+It is possible to prepare a simulation by explicitly passing in the instantiated objects you wish to use rather than
+getting them from a model specification file. To demonstrate this, we will recreate the simulation from the
+:ref:`disease_model.yaml <disease_model_yaml>` specification without using the actual file itself.
+
+Components
+~~~~~~~~~~
+
+We will first instantiate the components necessary for the simulation. In this case, we will get them directly from
+the example and we will place them in a list.
+
+.. code-block:: python
+
+    from vivarium.examples.disease_model import (BasePopulation, Mortality, Observer,
+                                                 SIS_DiseaseModel, Risk, DirectEffect,
+                                                 MagicWandIntervention)
+
+    components = [BasePopulation(),
+                  Mortality(),
+                  SIS_DiseaseModel('diarrhea'),
+                  Risk('child_growth_failure'),
+                  DirectEffect('child_growth_failure', 'infected_with_diarrhea.incidence_rate'),
+                  DirectEffect('child_growth_failure', 'infected_with_diarrhea.excess_mortality_rate'),
+                  MagicWandIntervention('breastfeeding_promotion', 'child_growth_failure.proportion_exposed'),]
+
+Configurations
+~~~~~~~~~~~~~~
+We also need to create a dictionary of the configurations for the components.
+
+.. code-block:: python
+
+    config = {
+       'randomness': {
+           'key_columns': ['entrance_time', 'age'],
+       },
+       'population': {
+           'population_size': 10_000,
+       },
+       'diarrhea': {
+           'incidence': 2.5,        # Approximately 2.5 cases per person per year.
+           'remission': 42,         # Approximately 6 day median recovery time
+           'excess_mortality': 12,  # Approximately 22 % of cases result in death
+       },
+       'child_growth_failure': {
+           'proportion_exposed': 0.5,
+       },
+       'effect_of_child_growth_failure_on_infected_with_diarrhea.incidence_rate': {
+           'relative_risk': 5,
+       },
+       'effect_of_child_growth_failure_on_infected_with_diarrhea.excess_mortality_rate': {
+           'relative_risk': 5,
+       },
+       'breastfeeding_promotion': {
+           'effect_size': 0.5,
+       },
+    }
+
+Initialize
+~~~~~~~~~~
+
+We can setup and advance the simulation as we did above. The object we get back, an :func:`InteractiveContext <vivarium.interface.interactive.InteractiveContext>`,
+is the same no matter which helper function you use, though it may or may not have had its setup function called. Here,
+we will use :func:`setup_simulation() <vivarium.interface.interactive.setup_simulation>` to automatically initialize and setup our
+simulation context.
+
+.. code-block:: python
+
+    from vivarium.interface import setup_simulation
+
+    sim = setup_simulation(components, config)
+    sim.step()
+
+.. _progressing:
+
+Advancing the Simulation
```

Comment:
Nice addition with this section. I really like having all of these in one table for quick reference/comparison